docs/contrib/quickstart.rst

```diff
@@ -0,0 +1,133 @@
+.. _quickstart:
+
+Quick Start Guide
+=================
+
+Requirements
+------------
+
+- Java 8 or newer
+- LLVM/Clang 15 or newer
+- sbt
+
+Project Structure Overview
+--------------------------
+
+See :ref:`build`
+
+Project suffix
+--------------
+
+Most projects in ScalaNative cross-build against Scala ``2.12``, ``2.13`` and ``3``, and these projects have a suffix like ``2_12``, ``2_13`` or ``3`` to differentiate the Scala version.
+For example, ``sandbox`` has ``sandbox2_12``, ``sandbox2_13`` and ``sandbox3``.
+
+In the following we will use suffix ``3``, but remember that you can build and test for different versions using different suffixes.
+
+Build / Manual Testing on Sandbox
+---------------------------------
+
+``sandbox3/run`` to compile, link and run the main method of the sandbox project defined in ``sandbox/src/main/scala/Test.scala``.
+
+It's convenient to run the ``sandbox`` project to verify the build works as expected.
+
+Test
+----
+
+**Common Test Commands**
+
+- ``tests3/test`` - run the unit tests for libraries on native build
+- ``tests3/testOnly org.scalanative.testsuite.javalib.util.RandomTest`` - run only the test of interest
+- ``tests3/testOnly *.RandomTest`` - run only the test of interest using wildcard
+- ``testsExt3/test`` - run the unit tests on native build, this module contains tests that requires dummy javalib implementation defined in ``javalibExtDummies``.
+- ``nirJVM3/test`` - run the unit tests for NIR
+- ``toolsJVM3/test`` - run the unit tests of the tools: ScalaNative backend
+- ``sbtScalaNative/scripted`` - run all `scripted tests <https://www.scala-sbt.org/1.x/docs/Testing-sbt-plugins.html>`_ of the sbt plugin (this takes a while).
+- ``sbtScalaNative/scripted <test directory to run>`` - run specific scripted tests of the sbt plugin. e.g. ``sbtScalaNative/scripted run/backtrace``
+    - Scripted tests are used when you need to interact with the file system, networking, or the build system that cannot be done with a unit test.
+    - ``set ThisBuild / scriptedBufferLog := false`` disables buffer log in scripted test and get more verbose output
+
+**Other Test Commands**
+
+- ``testsJVM3/test`` - run ``tests3/test`` on JVM
+- ``testsExtJVM3/test`` - run ``testsExt3/test`` on JVM
+- ``test-all`` - run all tests, ideally after ``reload`` and ``clean``
+  - Note: it is possible that some tests ran by `test-all` might not pass or compile on your system. We are currently working on stabilizing this test suite for our supported platforms
```

Comment:
I recommend deleting this section. Two recently merged PR have made `test-all` more robust.